pkg/controller/direct/cloudbuild/workerpool_controller.go

```diff
@@ -283,6 +287,8 @@ func (a *Adapter) Update(ctx context.Context, u *unstructured.Unstructured) erro
 	}
 	status.ObservedState.CreateTime = ToOpenAPIDateTime(updated.GetCreateTime())
 	status.ObservedState.UpdateTime = ToOpenAPIDateTime(updated.GetUpdateTime())
+	// This value should not be updated. Just in case.
+	status.ExternalRef = NewResourceRef(updated).Get()
```

Comment:
You may want to write a quick design-doc for "status.externalRef" including "why is it not status.selfLink", how it fixes resource deletion etc